src/main/java/org/apache/commons/lang3/Validate.java

```diff
@@ -874,7 +888,752 @@ public static void matchesPattern(final CharSequence input, final String pattern
+     * 
+     * @param max  the reference value
+     * @param value  the value to validate
+     * @param message  the {@link String#format(String, Object...)} exception message if invalid, not null
+     * @param values  the optional values for the formatted exception message
+     * @throws IllegalArgumentException if {@code value} is greater than {@code max}
+     * @see #smallerOrEqual(long, long) 
+     * 
+     * @since 3.4
+     */
+    public static void smallerOrEqual(final long max, final long value, final String message, final Object... values) {
+        if (value > max) {
+            throw new IllegalArgumentException(String.format(message, values));
+        }
+    }
+    
+    /**
+     * <p>Validates that the specified argument is smaller than, or equal to, a 
+     * given reference; otherwise throwing an exception.</p>
+     * 
+     * <p>If {@code min} or {@code value} is {@code NaN}, the test will fail and
+     * the exception will be thrown.</p>
+     * 
+     * <pre>Validate.smallerOrEqual(myDouble, 0.0);</pre>
+     * 
+     * <p>The message of the exception is &quot;The value {@code value} is not 
+     * smaller than or equal to {@code max}&quot;.</p>
+     * 
+     * @param max  the reference value
+     * @param value  the value to validate
+     * @throws IllegalArgumentException if {@code value} is greater than {@code max}
+     * @see #smallerOrEqual(double, double, java.lang.String, java.lang.Object...) 
+     * 
+     * @since 3.4
+     */
+    public static void smallerOrEqual(final double max, final double value) {
+        smallerOrEqual(max, value, DEFAULT_SMALLER_OR_EQUAL_EX_MESSAGE, value, max);
+    }
+    
+    /**
+     * <p>Validates that the specified argument is smaller than, or equal to, a 
+     * given reference; otherwise throwing an exception with the specified message.</p>
+     * 
+     * <p>If {@code min} or {@code value} is {@code NaN}, the test will fail and
+     * the exception will be thrown.</p>
+     * 
+     * <pre>Validate.smallerOrEqual(myDouble, 0.0);</pre>
+     * 
+     * @param max  the reference value
+     * @param value  the value to validate
+     * @param message  the {@link String#format(String, Object...)} exception message if invalid, not null
+     * @param values  the optional values for the formatted exception message
+     * @throws IllegalArgumentException if {@code value} is greater than {@code max}
+     * @see #smallerOrEqual(double, double) 
+     * 
+     * @since 3.4
+     */
+    public static void smallerOrEqual(final double max, final double value, final String message, final Object... values) {
+        if (!(value <= max)) {
+            throw new IllegalArgumentException(String.format(message, values));
+        }
+    }
+    
+    // different
+    //---------------------------------------------------------------------------------
+    
+    /**
+     * <p>Validates that the specified argument is not equal to a given value
```

Comment:
The wording here is a bit confusing since "not equal" usually means ```value.equals(reference) == false```. Maybe ```smallerOrGreater``` would be a better method name?